samples/tiled-convolution/tiled-convolution.cpp

```diff
@@ -0,0 +1,310 @@
+            cgh, cl::sycl::range<2>(mat_size.m, mat_size.n),
+            cl::sycl::id<2>(host_offset_m, host_offset_n));
+        // global size m
+        auto global_size_m = round_up(mat_size.m, ocl_config_t::local_size_m);
+        // global size n
+        auto global_size_n = round_up(mat_size.n, ocl_config_t::local_size_n);
+        // constructing the kernel
+        cgh.parallel_for(cl::sycl::nd_range<2>(
+                             cl::sycl::range<2>(global_size_m, global_size_n),
+                             cl::sycl::range<2>(ocl_config_t::local_size_m,
+                                                ocl_config_t::local_size_n),
+                             cl::sycl::id<2>(host_offset_m, host_offset_n)),
+                         conv<read_accessor_t, write_accessor_t, index_t,
+                              matrix_size_t<index_t>>(fil_acc, in_acc, out_acc,
+                                                      total_buffer, fil_size));
+      });
+// this part is used to profile each tiled kernel
+#ifdef PROFILE_SYCL
+      auto i = n + m * num_host_tile_n;
+      event.wait();
+      auto end = std::chrono::system_clock::now();
+      std::chrono::duration<double, std::milli>
+          current_application_execution_time = end - start;
+      total_application_execution_time +=
+          current_application_execution_time.count();
+      auto submit_time = event.get_profiling_info<
+          cl::sycl::info::event_profiling::command_submit>();
+      auto start_time = event.get_profiling_info<
+          cl::sycl::info::event_profiling::command_start>();
+      auto end_time = event.get_profiling_info<
+          cl::sycl::info::event_profiling::command_end>();
+      auto current_submission_time = (start_time - submit_time) / 1000000.0f;
+      auto current_execution_time = (end_time - start_time) / 1000000.0f;
+
+      total_execution_time += current_execution_time;
+      total_submission_time += current_submission_time;
+#ifdef PER_EVENT_PROFILING
+      std::cout << "event, " << i << " , current_kernel_submission_time, "
+                << current_submission_time
+                << " , current_kernel_execution_time, "
+                << current_execution_time
+                << ", current_application_execution_time, "
+                << current_application_execution_time.count() << "\n";
+#endif
+#endif
+      host_offset_n += mat_size.n;
+    }
+    host_offset_m += mat_size.m;
+  }
+// this part is used to profile the total execution time
+#ifdef PROFILE_SYCL
+  avarage_submission_time =
+      total_submission_time / float(num_host_tile_n * num_host_tile_m);
+  avarage_execution_time =
+      total_execution_time / float(num_host_tile_n * num_host_tile_m);
+  avarage_application_execution_time = total_application_execution_time /
+                                       float(num_host_tile_n * num_host_tile_m);
+  std::cout << " local_size_0, " << ocl_config_t::local_size_m
```

Comment:
This is also just very untidy. I understand we want to show how to profile your SYCL applications, but I don't think this is the way to do it.